### PR TITLE
Configure cell magics

### LINKED ISSUE #195: ENH make it configurable which cells to skip
ATM it's just bash and script, these could remain the default

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #195

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -51,3 +51,20 @@ or you can put the following in your :code:`.nbqa.ini` file
...
+.. code-block:: bash
+
+    nbqa black my_notebook.ipynb --nbqa-ignore-cells %%html,%%cython
+
+, or the :code:`ignore_cells` option in your :code:`.nbqa.ini` file, e.g.
+
+.. code-block:: ini
+
+    [black]
+    ignore_cells = 1
```
Comment (girip11) [inline docs/configuration.rst:70]: How do we specify `--nbqa-ignore-cells %%html,%%cython` in the `.ini` or any config file? 

```ini
    [black]
    ignore_cells = 1
```
Above config snippet in the documentation should be corrected to

```ini
    [black]
    ignore_cells = %%html,%%cpython
```

File: nbqa/save_source.py [hunk 2]
```diff
@@ -27,21 +28,32 @@ def _replace_magics(source: List[str]) -> Iterator[str]:
...
+        Extra cells which nbqa should ignore.
 
     Yields
     ------
     str
         Line from cell, possibly commented out.
     """
-    bash_cell = source and any(source[0].startswith(i) for i in ("%%script", "%%bash"))
+    if ignore_cells is not None:
+        ignore = MAGIC + ignore_cells.split(",")
```
Comment (girip11) [inline nbqa/save_source.py:40]: After strip, its better to strip leading and trailing whitespaces

Some users might provide `ignore_cells = %%html, %%cpython` in the configuration.

**Review by girip11 — CHANGES_REQUESTED** [review]

File: nbqa/save_source.py (hunk 2 above)
Comment (MarcoGorelli) [reply nbqa/save_source.py:40]: I hadn't thought of that, thanks! This makes it a bit more user-friendly, and I can't see any drawbacks...

File: docs/configuration.rst (hunk 1 above)
Comment (MarcoGorelli) [reply docs/configuration.rst:70]: That's a good point, thanks!

File: nbqa/save_source.py (hunk 2 above)
Comment (MarcoGorelli) [reply nbqa/save_source.py]: @girip11 done, thanks for the idea!

Let's see if this works:
@all-contributors please add @girip11 for ideas

**Comment by MarcoGorelli** [comment]
@all-contributors please add @girip11 for ideas

**Review by girip11 — APPROVED** [review]

File: docs/configuration.rst [hunk 3]
```diff
@@ -50,5 +50,22 @@ or you can put the following in your :code:`setup.cfg` file
 
 .. code-block:: cfg
 
-    [nbqa.mutate]
-    black = 1
+    [black]
+    mutate = 1
```
Comment (MarcoGorelli) [inline docs/configuration.rst:54]: this should've been changed

File: docs/configuration.rst [hunk 4]
```diff
@@ -50,5 +50,22 @@ or you can put the following in your :code:`setup.cfg` file
...
+.. code-block:: bash
+
+    nbqa black my_notebook.ipynb --nbqa-ignore-cells %%html,%%cython
+
+, or the :code:`ignore_cells` option in your :code:`.nbqa.ini` file, e.g.
+
+.. code-block:: ini
+
+    [black]
+    ignore_cells = %%html,%%cython
```
Comment (MarcoGorelli) [inline docs/configuration.rst:71]: change to reference setup.cfg

**Comment by MarcoGorelli** [comment]
Thanks for your review, merging then following approval

**Pull request merged**